### ISSUE TITLE
Navigate page till rendering last frame not networkidel

### ISSUE THREAD

**Issue by OmarAbdElNasser** [puppeteer/puppeteer#1500]
I'm taking screenshot for large html 6200*6000 , but i got large white space in the output image every time i try to take the image.

I made so many tests, and found out that there're frames rendered after the `load` event, so i don't get it in the captured image.

The red line showing the `load` event and there's frames after that line.

I tried to read all the documentation for the package but i'm a civil engineer and don't know lots in programming, i tried my best and couldn't find a solution for it.

There's 4 options in `page.goto` and they don't solve the problem, here's the url for the documentation .
https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagegotourl-options

Here's my code 

    const puppeteer = require('puppeteer');
    
    (async () => {
      const browser = await puppeteer.launch();
      const page = await browser.newPage();
      await page.goto('D:\Image_2.HTML', {waitUntil: 'load'});
      function sleep(ms) {
      return new Promise(resolve => setTimeout(resolve, ms));
    }
    console.log(page.frames())
      await page.screenshot({path: 'example.png',fullPage : true});
    
      await browser.close();
    })();

Help please.

**Comment by vsemozhetbyt** [comment]
@OmarAbdElNasser It is hard to test without the html page code. Can you provide the `Image_2.HTML` somehow?

By the way, the path should be the `'D:\\Image_2.HTML'` — path delimiter needs to be escaped on Windows.

**Comment by OmarAbdElNasser** [comment]
Thanks for replying , here's my html code 
https://drive.google.com/file/d/1A6M7zcipODtzQVtlm2NPOrNaxi3JABnS/view

**Comment by OmarAbdElNasser** [comment]
I made so many tests to find out why this happen. the load event occur before finishing rendering the whole image, so if we can change waitUntil: 'load' to something like waitUntil: 'FinishRendering', it will solve the problem and i'm sure of this , the problem is that there's no option called FinishRendering, so if there's something detect the FinishingRendering instead of load, it will be great.

**Comment by vsemozhetbyt** [comment]
FWIW, I do have white spot with the code:
<details>
<summary>Big Image:</summary>

![example0](https://user-images.githubusercontent.com/10393198/33426725-7cbbda3a-d5cb-11e7-9387-d68f20bccb84.png)


</details><br>

If I use `await puppeteer.launch({ headless: false });` I have no white spots, but the image seems distorted:

<details>
<summary>Big image:</summary>

![example1](https://user-images.githubusercontent.com/10393198/33426781-b4d71c04-d5cb-11e7-9a2c-e4d1e02c3dc8.png)

</details><br>

So it seems the issue is not the load/rendering race, but the screenshot getting/saving, I am not sure how to fix this, if I am right.

**Comment by OmarAbdElNasser** [comment]
I checked if there's problem in the image or not,  but it seems complete and nothing missed, Thank you so much, if we find away to make it headless it would be great.

**Comment by vsemozhetbyt** [comment]
@OmarAbdElNasser But in the headfull image, the roads seem misplaced, shifted in some places.

**Comment by OmarAbdElNasser** [comment]
Please, May you tell me where is this roads in the image ? i checked alot and there is nothing shifted.

**Comment by OmarAbdElNasser** [comment]
@vsemozhetbyt I think you r right in saying that the problem is in writing the image not the rendering and load race

**Comment by vsemozhetbyt** [comment]
@OmarAbdElNasser You can compare the upper right corner in both images: it seems like the left map part is copied above the right part in the second (headfull) image:

![1](https://user-images.githubusercontent.com/10393198/33429254-beb3f734-d5d4-11e7-9e2d-aa7dc42fb502.png)

**Comment by OmarAbdElNasser** [comment]
Please may you generate another image and compare ? i made image using phantomjs and one with headless false and they r the same and i checked the roads and still the same.

**Comment by vsemozhetbyt** [comment]
@OmarAbdElNasser Maybe the cause is in my OS or hardware etc. Another try:

Windows 7 x64
puppeteer 1.0.0-rc with Chromium 64.0.3276.0
Display 1366 x 768

```js
'use strict';

const puppeteer = require('puppeteer');

(async () => {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();

  await page.goto('J:\\temp\\test.html', { waitUntil: 'load' });

  await page.screenshot({ path: 'J:\\temp\\test.png', fullPage: true });

  await browser.close();
})();
```

<details>
<summary>Headless image:</summary>

![test0](https://user-images.githubusercontent.com/10393198/33430223-743b5acc-d5d8-11e7-8916-2ca5c346bb9f.png)


</details><br>

```js
'use strict';

const puppeteer = require('puppeteer');

(async () => {
  const browser = await puppeteer.launch({ headless: false });
  const page = await browser.newPage();

  await page.goto('J:\\temp\\test.html', { waitUntil: 'load' });

  await page.screenshot({ path: 'J:\\temp\\test.png', fullPage: true });

  await browser.close();
})();
```

<details>
<summary>Headfull image:</summary>

![test1](https://user-images.githubusercontent.com/10393198/33430247-8cfc7596-d5d8-11e7-94ba-ca6168448e5b.png)


</details><br>

As you can see, in the second (headfull) image the right part is wrong, just copied from the left one.

**Comment by OmarAbdElNasser** [comment]
I use the same code as you but i get another image .
![example](https://user-images.githubusercontent.com/34121159/33430972-254d0e30-d5db-11e7-8f11-8f2ac0b7db1d.png)

**Comment by vsemozhetbyt** [comment]
What is your OS and puppeteer/Chromium version, display resolution?

**Comment by OmarAbdElNasser** [comment]
Windows 7 ultimate SP1
Latest version of puppeteer
62.0.3202.94  Chromium
Display 1440*900

**Comment by OmarAbdElNasser** [comment]
I solved the problem but using another packages and putting the script in my question in stackoverflow
https://stackoverflow.com/questions/47564602/navigate-page-till-rendering-last-frame-not-networkidel/47577503#47577503

**Comment by vonGameTheory** [comment]
I get best results as far as getting "finished" pages when using {waitUntil:['load','domcontentloaded','networkidle0']} -- waits for all of these events to fire, otherwise I've gotten tripped up in some cases where the page does several redirects before the final page, probably works better in most cases.  (note that 'domcontentloaded' isn't available on versions <= 0.13)
